test/t8n/t8n.cpp

```diff
@@ -0,0 +1,172 @@
+#include "../state/mpt_hash.hpp"
+#include "../state/rlp.hpp"
+#include "../statetest/statetest.hpp"
+#include <evmone/evmone.h>
+#include <nlohmann/json.hpp>
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <string_view>
+
+namespace fs = std::filesystem;
+namespace json = nlohmann;
+namespace tu = evmone::test::utils;
+using namespace evmone;
+
+static const auto NULL_HEXSTRING_256 = "0x" + std::string(512, '0');
+static const auto NULL_HEXSTRING_32 = "0x" + std::string(64, '0');
+static const auto NULL_HEXSTRING_20 = "0x" + std::string(40, '0');
+
+namespace evmone::state
+{
+/// Defines how to RLP-encode a Log. This is only needed to compute "logs hash".
+inline bytes rlp_encode(const Log& log)
+{
+    return rlp::encode_tuple(log.addr, log.topics, log.data);
+}
+}  // namespace evmone::state
+
+int main(int argc, const char* argv[])
```

Comment:
We should have a unit test for this (mostly for checking the input -> output mapping is unchanged). Question is: is it better to use a CLI test or make this into an internal library first?

Reply:
I would prefer the second. @chfast What do you think?